account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/MFAMethodsCreateHandler.java

```diff
@@ -79,11 +82,21 @@ private APIGatewayProxyResponseEvent mfaMethodsHandler(
             return generateApiGatewayProxyErrorResponse(400, ErrorResponse.ERROR_1001);
         }
 
-        var maybeUserProfile = dynamoService.getOptionalUserProfileFromPublicSubject(subject);
+        Optional<UserProfile> maybeUserProfile =
+                dynamoService.getOptionalUserProfileFromPublicSubject(subject);
         if (maybeUserProfile.isEmpty()) {
             return generateApiGatewayProxyErrorResponse(404, ErrorResponse.ERROR_1056);
         }
-        var userProfile = maybeUserProfile.get();
+        UserProfile userProfile = maybeUserProfile.get();
+
+        Map<String, Object> authorizerParams = input.getRequestContext().getAuthorizer();
+        if (PrincipalValidationHelper.principalIsInvalid(
+                userProfile,
+                configurationService.getInternalSectorUri(),
+                dynamoService,
+                authorizerParams)) {
+            return generateApiGatewayProxyErrorResponse(404, ErrorResponse.ERROR_1079);
```

Comment:
404 doesn't seem right to me here, maybe a 401 (unauthorized) is more appropriate?